### PR TITLE
Add support for EGL_DEBUG_KHR extension

### PR DESCRIPTION
**Pull request by lfrb**

### CODE REVIEW

**Comment by jrfonseca** [comment]
@lfrb please rebase this and all your other PRs on top of https://github.com/apitrace/apitrace/commit/764c1f1d95123d38fd3f6b814da36482da4d8eb4 so mac64 build can succeed.

File: dispatch/glproc.py [hunk 1]
```diff
@@ -573,6 +576,9 @@ def failFunction(self, function):
             print(r'    if (messageLog != 0 && bufsize > 0) *messageLog = 0;')
             print(r'    return 0;')
             return
+        if function.name.startswith('eglQueryDebugKHR'):
+            print(r'    if (value != 0) *value = 0;')
+            return
```
Comment (jrfonseca) [inline dispatch/glproc.py:581]: `eglQueryDebugKHR` returns a `EGLBoolean`.  You're missing a

```python
    print(r' return EGLFalse;')
```

Ditto for all the other functions that return values .

This should fix the MSVC builds.

File: specs/eglapi.py [hunk 2]
```diff
@@ -434,4 +463,9 @@ def GlFunction(*args, **kwargs):
     GlFunction(EGLBoolean, "eglBindWaylandDisplayWL", [(EGLDisplay, "dpy"), (WlDisplay, "display")]),
     GlFunction(EGLBoolean, "eglUnbindWaylandDisplayWL", [(EGLDisplay, "dpy"), (WlDisplay, "display")]),
     GlFunction(EGLBoolean, "eglQueryWaylandBufferWL", [(EGLDisplay, "dpy"), (WlResource, "buffer"), (EGLint_enum, "attribute"), Out(Pointer(EGLint), "value")], sideeffects=False),
+
+    # EGL_KHR_debug
+    GlFunction(EGLint, "eglDebugMessageControlKHR", [(EGLDEBUGPROCKHR, "callback"), (EGLDebugAttribs, "attrib_list")]),
+    GlFunction(EGLBoolean, "eglQueryDebugKHR", [(EGLint, "attribute"), (OpaquePointer(EGLAttrib), "value")]),
```
Comment (jrfonseca) [inline specs/eglapi.py:469]: `sideffects=False`

File: specs/eglapi.py [hunk 3]
```diff
@@ -434,4 +463,9 @@ def GlFunction(*args, **kwargs):
     GlFunction(EGLBoolean, "eglBindWaylandDisplayWL", [(EGLDisplay, "dpy"), (WlDisplay, "display")]),
     GlFunction(EGLBoolean, "eglUnbindWaylandDisplayWL", [(EGLDisplay, "dpy"), (WlDisplay, "display")]),
     GlFunction(EGLBoolean, "eglQueryWaylandBufferWL", [(EGLDisplay, "dpy"), (WlResource, "buffer"), (EGLint_enum, "attribute"), Out(Pointer(EGLint), "value")], sideeffects=False),
+
+    # EGL_KHR_debug
+    GlFunction(EGLint, "eglDebugMessageControlKHR", [(EGLDEBUGPROCKHR, "callback"), (EGLDebugAttribs, "attrib_list")]),
```
Comment (jrfonseca) [inline specs/eglapi.py:468]: sideeffects=False

**Review by jrfonseca — CHANGES_REQUESTED** [review]
Otherwise looks good.

File: dispatch/glproc.py (hunk 1 above)
Comment (jrfonseca) [reply dispatch/glproc.py:580]: Thanks for then update.  I mistyped, it should have been `return EGL_FALSE`.  I know 0 and `EGL_FALSE` is all the same from the compiler POV, but it's confusing from my POV.

**Comment by lfrb** [comment]
@jrfonseca Anything else you would make to change to get it merged? Thanks :)

**Comment by jrfonseca** [comment]
@lfrb I missed the update notification somehow.  Thank you for that.

**Pull request merged**